File: ruby/grains/grains.rb
module BookKeeping
  VERSION = 1
end

class Grains
  
  def self.square i
    raise ArgumentError if i < 1 || i > 64
    2**(i-1)
  end
  
  def self.total
    (0..63).map { |x| 2**x }.reduce(&:+)
  end
    
end

File: javascript/grains/grains.js

let bigInt = require('./big-integer');

class Grains {
  
  square(i) {
    if (i < 1 || i > 64) throw new Error('ArgumentError');
    return bigInt(2).pow(i-1).toString()
  }
  
  total() {
    return Array(64).fill(0).map((x,i)=>bigInt(2).pow(i)).reduce((a,b)=>bigInt(a).plus(b)).toString()
  }
}

module.exports = Grains;